app/search.py

```diff
@@ -0,0 +1,53 @@
+import os
+import requests
+from dotenv import load_dotenv
+
+load_dotenv()
+
+
+class YTSearch:
+    """
+    Formatting for retrieving youtube search results for a given query
+    """
+
+    def __init__(self, query):
+        self.params = {
+            "videoCategoryId": "10",
+            "part": "snippet",
+            "type": "video",
+            "q": query,
+            "videoEmbeddable": True,
+            "key": os.getenv("YOUTUBE_KEY"),
+        }
+        self.root_url = "https://www.googleapis.com/youtube/v3/search"
+        self.results = self.get_results()
+        self.matches = self.results['result']['items']
+
+    def get_results(self):
+        resp = requests.get(self.root_url, self.params)
+        youtube_results = resp.json()
+
+        return {"resp": resp, "result": youtube_results}
+
+
+class LastFMSearch:
+    """
+    Formatting for retrieving last.fm search results for a given query
+    """
+
+    def __init__(self, query):
+        self.params = {
+            "method": "track.search",
+            "format": "json",
+            "track": query,
+            "api_key": os.getenv("LASTFM_KEY"),
```

Comment:
It would be better to fetch this API key once when you setup the app and then inject it in the constructor. Imagine it's used in many places and you need to change how it's retrieved. When you hardcode inline like this you have to track down every single instance where you get the API key and change it. This is a bit tedious and error prone. 

You also needlessly read the api key from the env on every search which is probably a small perf hit. This was one of the reasons I suggested it would be better to instantiate this search object once with the app vs. per call to the search API. Then you'd only need to read it once.

Reply:
so would this mean basically adding something in the `create_app` function that reads it from the .env then? I'm not sure where I would store it or how to effectively inject it then, other than holding it as a global variable that's accessed by the class constructor.

Or just make like one object that loads the keys and read the keys from that object?

Reply:
Check out https://github.com/gothinkster/flask-realworld-example-app

Keep in mind the architecture they use is suitable for a very mature app and more fancy and heavy handed than the one I used in that app I showed you. They use flask extensions to solve this problem. Another less fancy option would look like this: (see how config is injected into the request handler which can then be used by any route

```
application = Flask(__name__, static_folder="../static-build/")
if ...:
    application.config.from_object(config.DefaultConfig)
else:
    application.config.from_object(config.DevConfig)

request_handler = core.build_request_handler(application.config)
```

Reply:
To answer your question though I'd read it from the env in the config

Reply:
ok that makes a lot of sense